src/app/services/inventory_services.py

```diff
@@ -17,4 +20,35 @@ def create_product(product_category_id, user_id,  product_code, title, value,bra
         return ('Item cadastrado com sucesso')
     
     except:
-        return ('Erro ao cadastrar item')
\ No newline at end of file
+        return ('Erro ao cadastrar item')
+
+
+def generate_user_data(user_id):
+    user = user_services.get_by_id(user_id) if user_id else None
+        
+    user_data = {
+        'id': user['id'] if user else None,
+        'name': user['name'] if user else 'Na empresa'
+    }
+    
+    return user_data
+
+
+def format_result(inventories):
+    for inventory in inventories:  
+        inventory['user_id'] = generate_user_data(inventory['user_id'])
+        
+    return inventories
+
+
+def get_inventories_by_name(name, page=None):
+    result = Inventory.query.filter(Inventory.title.ilike(f"%{name}%")).paginate(per_page=20, page=page)
```

Comment:
Talvez encontrar uma forma de pesquisa que não necessite passar o name com letra maiúscula no início ou apenas colocar um Title ou Captalize no para 'name' da função para ele entrar na query com a primeira letra maiúscula caso o usuario não o fizer.
Ex:
mouse não encontra produto, porém Mouse encontra.